### PR TITLE
Applying new Prettier defaults to files - batch #5

### PR DESCRIPTION
**Pull request by halocline**
#### What does this PR do?

Batch # 5

Incrementally applying [new Prettier defaults](https://prettier.io/docs/en/options.html#arrow-function-parentheses). Applying in batches for easier review.

Applying remaining Prettier syntax fixes to unblock PRs which are held up from review.

#### What testing has been done on this PR?

Manual review of syntax changes, plus commit test suite.

#### How should this be manually tested?

Manual review of syntax changes

#### Any background context you want to provide?

https://prettier.io/docs/en/options.html#arrow-function-parentheses

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

no

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

File: src/js/components/List/List.js [hunk 1]
```diff
@@ -166,23 +162,25 @@ const List = React.forwardRef(
...
+
     if (active >= 0) {
       let activeId;
       // We have an item that is 'focused' within the list. This could
       // be the list item or one of the up/down ordering buttons.
       // We need to figure out an id of the thing that will be shown as active
       if (onOrder) {
         // figure out which arrow button will be the active one.
-        const buttonId = (active % 2) ? 'MoveDown' : 'MoveUp';
+        const buttonId = active % 2 ? 'MoveDown' : 'MoveUp';
```
Comment (taysea) [inline src/js/components/List/List.js:177]: Interesting that the parentheses are lost in this case, just confirming that's part of the new version rules?

File: src/js/components/List/List.js [hunk 2]
```diff
@@ -232,10 +229,8 @@ const List = React.forwardRef(
             (onClickItem || onOrder) && data && data.length
               ? () => {
                   const min = onOrder ? 1 : 0;
-                  const max = onOrder ? (data.length * 2) - 2 : data.length - 1;
-                  setActive(
-                    active >= min ? Math.min(active + 1, max) : min,
-                  );
+                  const max = onOrder ? data.length * 2 - 2 : data.length - 1;
```
Comment (taysea) [inline src/js/components/List/List.js:232]: seems like removal of parentheses for calculations is consistent across files, but just want to confirm it is expected.

**Review by taysea — COMMENTED** [review]
I believe all looks good, just confirming one type of change.

File: src/js/components/List/List.js (hunk 1 above)
Comment (halocline) [reply src/js/components/List/List.js:177]: Confirmed. I believe this is done by the `eslint-plugin-prettier` and the rule no-mixed-operators. Lots of folks on Google wondering why their mathematical operations loose their parens...

File: src/js/components/List/List.js (hunk 2 above)
Comment (halocline) [reply src/js/components/List/List.js:232]: Confirmed. I believe this is done by the `eslint-plugin-prettier`.

**Review by taysea — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**